roles/baseline/tasks/ingress-nginx.yaml

```diff
@@ -8,6 +8,26 @@
     - install
     - update
 
+- name: Set the default INGRESS_NGINX_CHART_VERSION if not specified
+  block:
+    - name: Retreive K8s cluster information
+      community.kubernetes.k8s_cluster_info:
+        kubeconfig: "{{ KUBECONFIG }}"
+      register: cluster_info
+    - name: Set the default INGRESS_NGINX_CHART_VERSION based on K8s server minor version
+      set_fact:
+        INGRESS_NGINX_CHART_VERSION: "{{ ingressVersions.k8sMinorVersionFloor.api.chartVersion
+                                      if cluster_info.version.server.kubernetes.minor is version(ingressVersions.k8sMinorVersionFloor.value, 'ge')
+                                      else ingressVersions.k8sMinorVersionCeiling.api.chartVersion }}"
+    - name: Output INGRESS_NGINX_CHART_VERSION
+      debug:
+        msg: "{{ INGRESS_NGINX_CHART_VERSION }}"
+  when:
+    - (INGRESS_NGINX_CHART_VERSION == "UNSPECIFIED") or (not INGRESS_NGINX_CHART_VERSION)
```

Comment:
For the check here based on my comment above the check here could be:
```yaml
when:
  - INGRESS_NGINX_CHART_VERSION|length > 0
```
or looking at it definition
```yaml
when:
  - INGRESS_NGINX_CHART_VERSION is not defined
```
again based on your logic flow.

Reply:
After reading your comments made some updates in the latest push.


The logic is now, `INGRESS_NGINX_CHART_VERSION` default value is now "" rather than UNSPECIFIED. If the user does not set `INGRESS_NGINX_CHART_VERSION` in their var file the new when statement `INGRESS_NGINX_CHART_VERSION|length == 0` will detect this and the task will run. The task will Then set the `INGRESS_NGINX_CHART_VERSION` value based on K8s version.